### ISSUE TITLE
Unable to 'edit starters' in child project

### ISSUE THREAD

**Issue by shankarramr** [Microsoft/vscode-spring-initializr#110]
I am unable to use the 'edit starters' feature from inside the pom of the child project.

Steps:
Created a spring boot project via Spring Initializr
Opened pom -> right click -> Edit starters

Error:
Fail to edit starters. Error: Not a valid Spring Boot project.

VScode version: 1.31.0-insider
Spring Initializr version: 0.4.4

Note: I am able to do this from the parent project pom, but it does not work from the child project pom

**Comment by Eskibear** [comment]
Current logic to judge a valid spring-boot project is simple and naive. It simply parse the pom file, and the condition is:
```ts
parent.groupId == "org.springframework.boot" && parent.artifactId == "spring-boot-starter-parent"
```

Any simple and effective way to identify a spring-boot project?

**Comment by Eskibear** [comment]
I'm thinking about the right behavior when editing starters for a child project. If some starters are inherited from its parent, should we be able to edit them?

e.g.
#### parent
```xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>yanzh</groupId>
  <artifactId>parent</artifactId>
  <version>0.1</version>
  <packaging>pom</packaging>
  <name>parent</name>
  <parent>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-parent</artifactId>
    <version>2.1.2.RELEASE</version>
    <relativePath /> <!-- lookup parent from repository -->
  </parent>
  <dependencies>
    <!-- web stater -->
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-web</artifactId>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-test</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>
</project>
```

#### child
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>yanzh</groupId>
    <artifactId>parent</artifactId>
    <version>0.1</version>
    <relativePath>../</relativePath>
  </parent>
  <groupId>yanzh</groupId>
  <artifactId>demo</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <name>demo</name>
  <description>Demo project for Spring Boot</description>
  <properties>
    <java.version>1.8</java.version>
  </properties>
  <build>
    <plugins>
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
      </plugin>
    </plugins>
  </build>
</project>
```

**Comment by Eskibear** [comment]
The fix has been released in v0.4.5.
